### PR TITLE
test(keccak-air): add proptests for AIR constraints and trace generation

### PR DESCRIPTION
**Pull request by tcoratger**
## Summary

- Add property-based tests that verify all ~2000 AIR constraints are satisfied for random inputs — both single and batch traces. This was entirely missing before.
- Add 8 negative unit tests + 2 negative proptests that corrupt each column family and assert constraint rejection (soundness coverage).
- Add witness-level proptests: output vs reference permutation, preimage persistence, step flag rotation, batch vs individual equivalence.
- Add `RC_BITS` consistency test against the `RC` constants.
- Remove trivial `rc_value_bit` wrapper in favor of direct `RC_BITS[r][i]` access.
- Add `proptest` to dev-dependencies.

## Test plan

- [x] `cargo test -p p3-keccak-air` — 22 tests pass in ~1 s
- [x] `cargo check -p p3-keccak-air --examples` — examples still compile
- [x] No changes to existing test code (diff is purely additive for `generation.rs`)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: keccak-air/src/constants.rs [hunk 1]
```diff
@@ -156,6 +156,46 @@ const RC_BITS: [[u8; 64]; 24] = [
     ],
 ];
 
-pub(crate) const fn rc_value_bit(round: usize, bit_index: usize) -> u8 {
-    RC_BITS[round][bit_index]
```
Comment (tcoratger) [inline keccak-air/src/constants.rs:160]: I think that this function was a bit useless since we can directly use `RC_BITS` within the crate as `pub(crate)`

**Review by Nashtare — APPROVED** [review]
Thanks, LGTM
